Source/OCMock/OCMStubRecorder.m

```diff
@@ -51,44 +51,46 @@ - (OCMInvocationStub *)stub
 
 - (id)andReturn:(id)anObject
 {
-	[[self stub] addInvocationAction:[[[OCMObjectReturnValueProvider alloc] initWithValue:anObject] autorelease]];
-	return self;
+    Class returnValueProviderClass;
+    if (anObject == mockObject)
+    {
+        returnValueProviderClass = [OCMNonRetainingObjectReturnValueProvider class];
+    } else
+    {
+        returnValueProviderClass = [OCMObjectReturnValueProvider class];
+    }
+    [[self stub] addInvocationAction:[[[returnValueProviderClass alloc] initWithValue:anObject] autorelease]];
+    return self;
 }
 
 - (id)andReturnValue:(NSValue *)aValue
 {
     [[self stub] addInvocationAction:[[[OCMBoxedReturnValueProvider alloc] initWithValue:aValue] autorelease]];
-	return self;
-}
-
-- (id)andReturnMockObject
-{
-	[[self stub] addInvocationAction:[[[OCMNonRetainingObjectReturnValueProvider alloc] initWithValue:mockObject] autorelease]];
-	return self;
+	  return self;
```

Comment:
I am going to merge this PR as is. In future, though, please don't make whitespace changes. I know that inconsistencies have crept in over the years, but cleaning them up like this adds noise to the real change. Thanks.